docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,328 @@
+     )
+
+Structured Grids
+~~~~~~~~~~~~~~~~
+Assigning data to locations using a structured grid is essentially an act of
+matching coordinate arrays to each dimension of the data array. The data can
+also be represented as an area (instead of a point) by including a bounds array
+for each coordinate array.
+
+.. figure:: images/data_structured_grid.svg
+   :alt: Diagram of how data is represented on a structured grid
+
+   Data on a structured grid
+
+   :download:`full size image <images/data_structured_grid.svg>`
+
+Unstructured Meshes
+~~~~~~~~~~~~~~~~~~~
+The most basic element in a mesh is the 0-dimensional **node**: a single
+location in space. Every node in the mesh is defined by indexing the
+1-dimensional X and Y (and optionally Z) coordinate arrays (the
+``node_coordinates``) - e.g. ``(x[3], y[3])`` gives the position of the fourth
+node. Note that this means each node has its own coordinates, independent of
+every other node. Since nodes can be anywhere in the space - **unstructured** -
+the position in the array has nothing to do with spatial position.
+
+If data is assigned to node location it must be stored in a 1D array of equal
+length to the number of nodes in the mesh. ``data[3]`` is at the position:
+``(x[3], y[3])``.
+
+Data can also be assigned to higher dimensional elements - **edges**, **faces**
+or **volumes**. These elements are constructed by connecting nodes together
+using a 2-dimensional **connectivity** array. One dimension varies over each
+element, while the other dimension varies over each node that makes up that
+element; the values in the array are the node indices. E.g. we could make 2
+square faces from 6 nodes using this ``face_node_connectivity``:
+``[[0, 1, 3, 2], [2, 3, 5, 4]]``. Remember that meshes are **unstructured**, so
+there is no significance to the order of the faces in the array. Data assigned
+to a higher dimensional location must be stored in a 1D array of equal length
+to the number of elements in the mesh, e.g. ``my_data = [0.33, 4.02]`` for our
+example.
+
+.. note::
+
+        Connectivities also exist to connect the higher dimensional elements,
+        e.g. ``face_edge_connectivity``. These are optional conveniences to
+        speed up certain operations and will not be discussed here.
+
+.. figure:: images/data_ugrid_mesh.svg
+   :alt: Diagram of how data is represented on an unstructured mesh
+
+   Data on an Unstructured Mesh
+
+   :download:`full size image <images/data_ugrid_mesh.svg>`
+
+----
+
+The mesh model also supports edges/faces/volumes having associated 'centre'
+coordinates - to allow point data to be assigned to these elements. 'Centre' is
+just a convenience term - the points can exist anywhere within their respective
+elements.
+
+.. figure:: images/ugrid_element_centres.svg
+   :alt: Diagram demonstrating mesh face-centred data.
+
+   Data can be assigned to mesh edge/face/volume 'centres'
+
+Mesh Flexibility
```

Comment:
This feels different to the sections above - we're out of "what's going on" and into "why is it better". Maybe this moves into "What this means"?

Reply:
Are you thinking it would be better to rename this section to `Why use Meshes?` and promote it to the higher heading level?

Reply:
Scratch that, it definitely belongs in a subheading concerning `The Detail`. I never meant to imply that `The Detail` should be limited to "what's going on" - I just wanted to isolate the deep explanation so that those who want the 'cliff notes' can skip it.

Reply:
3c0e5f85bff8e008226dc9d8a88da47b9b6f4db1